src/module.ts

```diff
@@ -0,0 +1,123 @@
+import { defineNuxtModule, logger, addComponentsDir, isNuxt2 as _isNuxt2, addPluginTemplate } from '@nuxt/kit'
+import defu from 'defu'
+import { ADSENSE_URL, DEFAULTS, ModuleOptions, TEST_ID, CONFIG_KEY } from './config'
+import { resolveRuntimeDir, resolveTemplateDir } from './dirs'
+
+export type { ModuleOptions }
+
+export default defineNuxtModule<ModuleOptions>({
+  meta: {
+    name: '@nuxtjs/google-adsense',
+    configKey: CONFIG_KEY,
+    compatibility: {
+      nuxt: '^2.15.0 || ^3.0.0-rc.11'
+    }
+  },
+  defaults: (nuxt) => ({
+    ...DEFAULTS,
+    test: nuxt.options.dev && process.env.NODE_ENV !== 'production'
+  }),
+  setup (options, nuxt) {
+    if (options.test) {
+      logger.info('Test mode enabled - Using Test Adsense ID')
```

Comment:
might be worth using consistent 'adsense' case throughout

Reply:
Just checked; I think it is [AdSense](https://www.google.com/intl/nl_nl/adsense/start/).

Reply:
Alright, will incorporate that one then 😋

Reply:
sorry 💀